### PR TITLE
Chore: refactor credentials and cost

### PR DESCRIPTION
**Pull request by TomerHeber**
### Solution

Removed data_cost_credentials duplication.

I haven't refactored the data_cost_credentials_test.go on purpose (will be made in a follow up PR).
This is to show that "nothing has changed".

Part of #394.

### CODE REVIEW

File: env0/data_cost_credentials_test.go [hunk 1]
```diff
@@ -97,14 +97,14 @@ func TestUnitAzureCostCredentialsDataSource(t *testing.T) {
```
Comment (TomerHeber) [inline env0/data_cost_credentials_test.go:97]: Left the test as is (with the exception of error messages), this shows that the data_cost_credentials isn't required and can safely be deleted.

File: env0/provider.go [hunk 2]
```diff
@@ -55,9 +55,9 @@ func Provider(version string) plugin.ProviderFunc {
 				"env0_template":                dataTemplate(),
 				"env0_templates":               dataTemplates(),
 				"env0_ssh_key":                 dataSshKey(),
-				"env0_aws_cost_credentials":    dataCostCredentials(string(client.AwsCostCredentialsType)),
-				"env0_azure_cost_credentials":  dataCostCredentials(string(client.AzureCostCredentialsType)),
-				"env0_google_cost_credentials": dataCostCredentials(string(client.GoogleCostCredentialsType)),
+				"env0_aws_cost_credentials":    dataCredentials("aws_cost"),
```
Comment (TomerHeber) [inline env0/provider.go:58]: using the regular dataCredentials.

File: env0/data_credentials.go [hunk 3]
```diff
@@ -2,13 +2,21 @@ package env0
...
 import (
 	"context"
-	"strings"
 
 	"github.com/env0/terraform-provider-env0/client"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
 )
 
+var credentialsTypeToPrefixList map[string][]string = map[string][]string{
```
Comment (TomerHeber) [inline env0/data_credentials.go:11]: Based on the feedback I received in Slack. This is the correct mapping.
If I got anything wrong, please let me know.

File: env0/resource_aws_credentials.go [hunk 4]
```diff
@@ -126,7 +126,7 @@ func resourceAwsCredentialsDelete(ctx context.Context, d *schema.ResourceData, m
 }
 
 func resourceAwsCredentialsImport(ctx context.Context, d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
-	credentials, err := getCredentials(d.Id(), "AWS_", meta)
+	credentials, err := getCredentials(d.Id(), []string{"AWS_"}, meta)
```
Comment (TomerHeber) [inline env0/resource_aws_credentials.go:129]: the resources will also be refactored eventually.

File: env0/data_cost_credentials_test.go (hunk 1 above)
Comment (TomerHeber) [reply env0/data_cost_credentials_test.go:97]: will refactor in next PR...

**Comment by sagilaufer1992** [comment]
Why prefix list? it seems you give a singe-item list

**Comment by TomerHeber** [comment]
> Why prefix list? it seems you give a singe-item list

@sagilaufer1992 see:

```
	"gcp":        {string(client.GcpServiceAccountCredentialsType)},
	"azure":      {string(client.AzureServicePrincipalCredentialsType)},
	"aws":        {string(client.AwsAssumedRoleCredentialsType), string(client.AwsAccessKeysCredentialsType)},
	"gcp_cost":   {string(client.GoogleCostCredentialsType)},
	"azure_cost": {string(client.AzureCostCredentialsType)},
	"aws_cost":   {string(client.AwsCostCredentialsType)},
```

**Review by sagilaufer1992 — APPROVED** [review]

**Pull request merged**